File: typescript/412-fizz-buzz.ts
function fizzBuzz(n: number): string[] {
    const result: string[] = new Array(n)
    for (let i = 0; i < n; i++) {
        let num = i + 1
        const divisibleByThree = num % 3 === 0
        const divisibleByFive = num % 5 === 0
        if (divisibleByThree && divisibleByFive) {
            result[i] = 'FizzBuzz'
            continue
        }
        if (divisibleByThree) {
            result[i] = 'Fizz'
            continue
        }
        if (divisibleByFive) {
            result[i] = 'Buzz'
            continue
        }   
        result[i] = `${num}`
    }
    return result
};

File: golang/412-fizz-buzz.go
package golang

import "strconv"

func fizzBuzz(n int) []string {
    result := make([]string, n)
    for i := 0; i < n; i++ {
        num := i + 1
        divisibleByThree := num % 3 == 0
        divisibleByFive := num % 5 == 0
        if divisibleByThree && divisibleByFive {
            result[i] = "FizzBuzz"
            continue
        }
        if divisibleByThree {
            result[i] = "Fizz"
            continue
        }
        if divisibleByFive {
            result[i] = "Buzz"
            continue
        }
        result[i] = strconv.Itoa(num)
    }
    return result
}